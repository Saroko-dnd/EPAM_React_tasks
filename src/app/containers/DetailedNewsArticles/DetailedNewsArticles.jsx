import React from 'react';
import PropTypes from 'prop-types';
import Carousel from '../../components/Carousel';

import ArticlesList from '../../components/ArticlesList';
import { customPropsTypes } from '../../constants';

class DetailedNewsArticles extends React.Component {
    componentDidMount() {
        const { newsApiToken, newsApiLink } = this.props;
        const newsApiRequestLink = `${newsApiLink}country=us&apiKey=${newsApiToken}`;

        this.props.loadNews(newsApiRequestLink);
    }

    componentDidUpdate(prevProps) {
        if (
            prevProps.detailedArticles.length === 0 &&
            this.props.detailedArticles.length > 0
        ) {
            this.props.loadRelatedNews(0);
        }
    }

    render() {
        this.finishedRendering = true;
        return (
            <section className="d-flex align-items-center justify-content-center flex-column">
                <Carousel
                    onChange={(index) => {
                        this.props.loadRelatedNews(index);
                    }}
                    elements={this.props.detailedArticles}
                />
                <ArticlesList
                    title="Related news"
                    articles={this.props.relatedArticles}
                    linksToDetails={false}
                />
            </section>
        );
    }
}

DetailedNewsArticles.propTypes = {
    newsApiLink: PropTypes.string.isRequired,
    newsApiToken: PropTypes.string.isRequired,
    loadNews: PropTypes.func.isRequired,
    relatedArticles: PropTypes.arrayOf(customPropsTypes.article),
    loadRelatedNews: PropTypes.func.isRequired,
    detailedArticles: PropTypes.arrayOf(PropTypes.element),
};

DetailedNewsArticles.defaultProps = {
    relatedArticles: [],
    detailedArticles: [],
};

export default DetailedNewsArticles;
