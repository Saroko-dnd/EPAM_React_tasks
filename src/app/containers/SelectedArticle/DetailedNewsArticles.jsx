import React from 'react';
import PropTypes from 'prop-types';
import Carousel from '../../components/Carousel';

import ArticlesList from '../../components/ArticlesList';
import { customPropsTypes } from '../../constants';

class DetailedNewsArticles extends React.Component {
    componentDidMount() {
        this.props.loadRelatedNews(0);
    }

    render() {
        this.finishedRendering = true;
        return (
            <section className="d-flex align-items-center justify-content-center flex-column">
                <Carousel
                    onChange={(index) => {
                        this.props.loadRelatedNews(index);
                    }}
                    elements={this.props.detailedArticles}
                />
                <ArticlesList
                    title="Related news"
                    articles={this.props.relatedArticles}
                    linksToDetails={false}
                />
            </section>
        );
    }
}

DetailedNewsArticles.propTypes = {
    relatedArticles: PropTypes.arrayOf(customPropsTypes.article),
    loadRelatedNews: PropTypes.func.isRequired,
    detailedArticles: PropTypes.arrayOf(PropTypes.element).isRequired,
};

DetailedNewsArticles.defaultProps = {
    relatedArticles: [],
};

export default DetailedNewsArticles;
