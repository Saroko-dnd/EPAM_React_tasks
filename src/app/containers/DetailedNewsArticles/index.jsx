import { connect } from 'react-redux';

import actions from '../../actions';
import DetailedNewsArticles from './DetailedNewsArticles';
import { actionTypes } from '../../constants';
import { getUniqueRelatedArticles, getDetailedArticles } from '../../selectors';

const mapStateToProps = state => ({
    relatedArticles: getUniqueRelatedArticles(state),
    detailedArticles: getDetailedArticles(state),
    selectedArticleIndex: state.selectedArticleIndex,
    dataDownloaded: !state.newsIsLoading,
});

const mapDispatchToProps = dispatch => ({
    loadRelatedNews: (index) => {
        dispatch(actions.loadRelatedNews(index));
    },
    loadNews: apiUrl =>
        dispatch(actions.loadNews(apiUrl, actionTypes.TOP_NEWS_DOWNLOADED)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DetailedNewsArticles);
