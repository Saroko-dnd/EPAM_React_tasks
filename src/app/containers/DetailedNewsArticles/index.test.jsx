import DetailedNewsArticles from './DetailedNewsArticles';

describe('render an DetailedNewsArticles container component', () => {
    const loadNewsCallback = jest.fn();
    const loadRelatedNewsCallback = jest.fn();
    const detailedNewsArticlesWrapper = shallow(<DetailedNewsArticles
        loadNews={loadNewsCallback}
        loadRelatedNews={loadRelatedNewsCallback}
        newsApiLink="link"
        newsApiToken="token"
    />);

    it('should render DetailedNewsArticles correctly', () => {
        expect(detailedNewsArticlesWrapper).toMatchSnapshot();
    });
});
