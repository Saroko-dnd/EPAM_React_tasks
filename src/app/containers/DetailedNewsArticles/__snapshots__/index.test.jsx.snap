// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an DetailedNewsArticles container component should render DetailedNewsArticles correctly 1`] = `
<section
  className="d-flex align-items-center justify-content-center flex-column"
>
  <CustomCarousel
    elements={Array []}
    onChange={[Function]}
  />
  <LoadingIndicator
    color="#009900"
    message="Loading related news..."
    size={150}
  />
</section>
`;
